src/components/Signup/SignupForm.js

```diff
@@ -0,0 +1,129 @@
+import { useState } from "react"
+import { Form, Input, Label, Button, Text } from "../Reusable"
+import { getSolidBtnTextColor, getSolidBtnBgColor, timer, emailIsInvalid, nameIsInvalid, passIsInvalid, passAndConfPassAreDiff } from "../../utils"
+import { useTheme } from '../../contexts/theme.context'
+
+const SignupForm = () => {
+    const [enteredEmail, setEnteredEmail] = useState('')
+    const [emailError, setEmailError] = useState(false)
+
+    const [enteredFName, setEnteredFName] = useState('')
+    const [fNameError, setFNameError] = useState(false)
```

Comment:
You can use reducer here instead of state.

Reply:
thanks for pointing out. i have made that change in another pr.